source/core/src/main/com/csse3200/game/components/player/PlayerStatsDisplay.java

```diff
@@ -57,11 +56,44 @@ public void create() {
 
         entity.getEvents().addListener("updateHealth", this::updatePlayerHealthUI);
         entity.getEvents().addListener("melee_pickup", this::updateMeleeWeaponUI);
-        entity.getEvents().addListener("ranged_pickup", this::updateRangedWeaponUI);
+        entity.getEvents().addListener("ranged_pickup", (Integer maxAmmo) -> {
+            updateShotgunLabel(20, "Shotgun");
+            updateRangedWeaponUI(maxAmmo, "Shotgun");
+        });
         entity.getEvents().addListener("ranged_activate", this::updateAmmoDisplay);
         entity.getEvents().addListener("updateSpeedPercentage", this::updateSpeedPercentageUI);
         entity.getEvents().addListener("updateDamageBuff", this::updateDamageUI);
         entity.getEvents().addListener("updateSpeedUI", this::updateSpeedPercentageUI);
+
+        initializeWeaponsAndAmmo();
+    }
+
+    /**
+     * Initializes the weapon and ammo counts based on the player selected.
+     * Retrieves the player type from PlayerAnimationController.
+     */
+    private void initializeWeaponsAndAmmo() {
```

Comment:
This is not really needed, you can just check for the player's inventory for weapons.